xcm/src/v0/multi_location.rs

```diff
@@ -367,6 +367,31 @@ impl MultiLocation {
 		return self.at(prefix.len())
 	}
 
+	/// Returns whether `self` begins with or is equal to `prefix`.
+	///
+	/// # Example
+	/// ```rust
+	/// # use xcm::v0::{Junction::*, MultiLocation::*};
+	/// # fn main() {
+	/// let m = X4(Parent, PalletInstance(3), OnlyChild, OnlyChild);
+	/// assert!(m.matches_prefix(&X2(Parent, PalletInstance(3))));
+	/// assert!(m.matches_prefix(&m));
+	/// assert!(!m.matches_prefix(&X2(Parent, GeneralIndex(99))));
+	/// assert!(!m.matches_prefix(&X1(PalletInstance(3))));
+	/// # }
+	/// ```
+	pub fn matches_prefix(&self, prefix: &MultiLocation) -> bool {
+		if self.len() < prefix.len() {
+			return false
+		}
+		for i in 0..prefix.len() {
```

Comment:
```suggestion
		prefix.iter().zip(self.iter()).all(|l, r| l == r)
```

This could be used to replace the rest of the function

Reply:
That's how I implemented it in Cumulus, but here I figured I'd keep to the imperative style of `match_and_split` :smile: 
Can definitely do.